libtelemetry/src/androidTest/java/com/mapbox/android/telemetry/ConfigurationClientInstrumentationTest.java

```diff
@@ -0,0 +1,57 @@
+package com.mapbox.android.telemetry;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.support.test.InstrumentationRegistry;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import okhttp3.Call;
+import okhttp3.OkHttpClient;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+public class ConfigurationClientInstrumentationTest {
```

Comment:
are we actually hitting real backend with this instrumentation test?

Reply:
We are currently not hitting real backend. Spent 2+ hours working on solution and decided to get the ball rolling on the review process. Did not find solution to this amongst coworkers and online resources. `CountdownLatch`, which I used previously for similar Async tests is also not working in this situation. 

Please advise or point in right direction for hunting down a good solution on this.

Reply:
would this be helpful: https://developer.android.com/reference/androidx/test/espresso/idling/net/UriIdlingResource ?